server_environment/README.rst

```diff
@@ -0,0 +1,208 @@
+* you can override or extend values in the main configuration
+  file of your instance;
+
+Environment variable
+--------------------
+
+You can define configuration in the environment variable ``SERVER_ENV_CONFIG``
+and/or ``SERVER_ENV_CONFIG_SECRET``. The 2 variables are handled the exact same
+way, this is only a convenience for the deployment where you can isolate the
+secrets in a different, encrypted, file. They are multi-line environment variables
+in the same configparser format than the files.
+If you used options in ``server_environment_files``, the options set in the
+environment variable overrides them. 
+
+The options in the environment variable are not dependent of ``running_env``,
+the content of the variable must be set accordingly to the running environment.
+
+Example of setup:
+
+A public file, containing that will contain public variables::
+
+    # These variables are not odoo standard variables,
+    # they are there to represent what your file could look like
+    export WORKERS='8'
+    export MAX_CRON_THREADS='1'
+    export LOG_LEVEL=info
+    export LOG_HANDLER=":INFO"
+    export DB_MAXCONN=5
+
+    # server environment options
+    export SERVER_ENV_CONFIG="
+    [storage_backend.my-sftp]
+    sftp_server=10.10.10.10
+    sftp_login=foo
+    sftp_port=22200
+    directory_path=Odoo
+    "
+
+A second file which is encrypted and contains secrets::
+
+    # This variable is not an odoo standard variable,
+    # it is there to represent what your file could look like
+    export DB_PASSWORD='xxxxxxxxx'
+    # server environment options
+    export SERVER_ENV_CONFIG_SECRET="
+    [storage_backend.my-sftp]
+    sftp_password=xxxxxxxxx
+    "
+
+Default values
+--------------
+
+When using the ``server.env.mixin`` mixin, for each env-computed field, a
+companion field ``<field>_env_default`` is created. This field is not
+environment-dependent. It's a fallback value used when no key is set in
+configuration files / environment variable.
+
+When the default field is used, the field is made editable on Odoo.
+
+Note: empty environment keys always take precedence over default fields
+
+
+Keychain integration
+--------------------
+
+Read the documentation of the class `models/server_env_mixin.py
+<models/server_env_mixin.py>`_.
+
+
+Usage
+=====
+
+You can include a mixin in your model and configure the env-computed fields
+by an override of ``_server_env_fields``.
+
+::
+
+    class StorageBackend(models.Model):
+        _name = "storage.backend"
+        _inherit = ["storage.backend", "server.env.mixin"]
+
+        @property
+        def _server_env_fields(self):
+            return {"directory_path": {}}
+
+Read the documentation of the class and methods in `models/server_env_mixin.py
+<models/server_env_mixin.py>`__.
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/149/10.0
```

Comment:
:fire: can you move README to new structure? :wink:

Reply:
Yes of course. Done.